opentracing-api/src/span.rs

```diff
@@ -0,0 +1,51 @@
+use SpanContext;
+use TagValue;
+
+/// The `Span` represents the OpenTracing specification's Span contract.
+pub trait Span<'a> {
+    /// The associated `SpanContext`.
+    type Context: SpanContext<'a>;
+
+    /// Retrieve the associated `SpanContext`.
+    ///
+    /// This may be called any time, including after `finish`.
+    fn context(&self) -> &'a Self::Context;
```

Comment:
The SpanContext should be Clone or returned owned is that it will be used for references (child of/follows from). -> if thats the case then I think we should add the clone constraint so users can clone if needed? .. let's see how it turns out down the road once we do more impls

Reply:
I agree it is best if we mark `SpanContext: Clone` as that gives more flexibility to users:+1: